pyqtgraph/tests/image_testing.py

```diff
@@ -0,0 +1,520 @@
+# Image-based testing borrowed from vispy
+
+"""
+Procedure for unit-testing with images:
+
+1. Run unit tests at least once; this initializes a git clone of
+   pyqtgraph/test-data in ~/.pyqtgraph.
+
+2. Run individual test scripts with the PYQTGRAPH_AUDIT environment variable set:
+
+       $ PYQTGRAPH_AUDIT=1 python pyqtgraph/graphicsItems/tests/test_PlotCurveItem.py
+
+   Any failing tests will
+   display the test results, standard image, and the differences between the
+   two. If the test result is bad, then press (f)ail. If the test result is
+   good, then press (p)ass and the new image will be saved to the test-data
+   directory.
+
+3. After adding or changing test images, create a new commit:
+
+        $ cd ~/.pyqtgraph/test-data
+        $ git add ...
+        $ git commit -a
+
+4. Look up the most recent tag name from the `testDataTag` variable in
+   getTestDataRepo() below. Increment the tag name by 1 in the function
+   and create a new tag in the test-data repository:
+
+        $ git tag test-data-NNN
+        $ git push --tags origin master
+
+    This tag is used to ensure that each pyqtgraph commit is linked to a specific
+    commit in the test-data repository. This makes it possible to push new
+    commits to the test-data repository without interfering with existing
+    tests, and also allows unit tests to continue working on older pyqtgraph
+    versions.
+
+    Finally, update the tag name in ``getTestDataRepo`` to the new name.
```

Comment:
Actually both instructions are wrong!  The testDataTag is a module-level global.